src/qibolab/instruments/qblox/port.py

```diff
@@ -227,15 +217,11 @@ def lo_enabled(self, value):
         self._settings.lo_enabled = value
         if self.module.device:
             if self.module.device.is_qrm_type:
-                self.module._set_device_parameter(
-                    self.module.device,
-                    f"out{self.port_number}_in{self.port_number}_lo_en",
-                    value=value,
+                self.module.device.set(
+                    f"out{self.port_number}_in{self.port_number}_lo_en", value=value
                 )
             elif self.module.device.is_qcm_type:
-                self.module._set_device_parameter(
-                    self.module.device, f"out{self.port_number}_lo_en", value=value
-                )
+                self.module.device.set(f"out{self.port_number}_lo_en", value=value)
 
     @property
     def lo_frequency(self):
```

Comment:
This comment actually refers to a couple of lines below, but I cannot comment there because it is not part of this PR. It is also relevant for all other properties, not only `lo_frequency`.

Here, if you are connected (`if self.module.device is not None`) you query the device (`device.get`) for the corresponding value. In principle you can just return `self._settings.lo_frequency` directly even when you are connected since, if things are done properly, this value should always agree with the one uploaded in the instrument. If there is a situation where these two values are different then the user probably misused the driver (ie. accessed the qcodes interface directly, bypassing qibolab) or there is a bug.

This is the approach we follow in the local oscillators. It could also be slightly faster to return a value stored in the host memory than querying the instrument.

Reply:
Opened issue about it, thanks.